testng/src/main/java/cucumber/api/testng/TestNGCucumberRunner.java

```diff
@@ -68,6 +69,10 @@ public void runCucumber(CucumberFeature cucumberFeature) {
         }
     }
 
+    public void runScenario(PickleEvent pickle) throws Throwable {
+        runtime.runPickle(pickle);
```

Comment:
So I did succeed in pushing a commit onto your pull request branch with the result listener and the update of `runScenario`, very convenient.